### ISSUE TITLE
`cell-var-from-loop` false positive for keyword-only arguments

### ISSUE THREAD

**Issue by olmokramer** [PyCQA/pylint#5012]
### Bug description

```python
# pylint: disable=missing-docstring,invalid-name
for i in range(5):
    def f(*, _i=i):
        print(_i)
    f()
    def g(_i=i):
        print(_i)
    g()
```

### Command used

```shell
pylint pylint_repro.py
```

### Pylint output

```shell
************* Module pylint_repro
pylint_repro.py:3:16: W0640: Cell variable i defined in loop (cell-var-from-loop)
```

### Expected behavior

I would expect the `cell-var-from-loop` not to be triggered because the value is bound to a keyword argument. The message is only triggered in the presence of the `*` that forces all following arguments to be keyword-only, i.e. it is triggered for `f` but not for `g`.

### Pylint version

```shell
pylint 2.10.2
astroid 2.7.3
Python 3.9.2 (default, Mar 15 2021, 17:11:34)
```

### OS / Environment

Ubuntu 20.04.3 LTS, Python installed with pyenv.
